src/main/drivers/system.c

```diff
@@ -216,6 +230,14 @@ void initialiseMemorySections(void)
     memcpy(&tcm_code_start, &tcm_code, (size_t) (&tcm_code_end - &tcm_code_start));
 #endif
 
+#ifdef USE_CCM_CODE
```

Comment:
Maybe it's time to use linker-created table for copying memory ..

Reply:
this intrigues me, do you have any links that I can read up on and/or examples of this?  Probably out of scope for this PR though, but if the idea is good and workable we can make another PR for it.

Reply:
Can't find anything quickly.
The idea is to create table of <src,dst,length> using linker script. Then this initialization code will become loop over the table. 
Only table start symbol needs to be exported.
End of table may be encoded as NULL or possibly using some bits in length to mark last entry.

Reply:
One possibility to extend this idea is to add compression, allowing to store .data (and possibly default config) in compressed format. For example LZ4 decoder is trivial (and can be simplified further for this purpose), so it will probably result in considerable FLASH saving ...

One problem with this is how to allocate compressed image in FLASH - unless it is last in FLASH, it will create interesting problems (linker needs to know compressed size / changing size may move symbols and modify compression ratio)

Reply:
yeah, those are good ideas for another PR.  I'll resolve this conversation, but if you find anything more on this maybe open an issue in the issue tracker and give me a mention on it.  I'm all for not wasting flash space and simplifying copy operations.  One of the things that we could also do is ditch some of the existing assembler startup code that does similar operations and even store the data sections compressed!